src/shell/shell_cmds.cpp

```diff
@@ -2242,3 +2243,298 @@ void DOS_Shell::CMD_VOL(char* args)
+	}
+
+	cmd_move_arguments parsed = cmd_move_parse_arguments(args);
+	if (parsed.error) {
+		WriteOut(MSG_Get(parsed.error));
+		return;
+	}
+
+	char temp[DOS_PATHLENGTH];
+	if (!DOS_Canonicalize(parsed.destination.c_str(), temp)) {
+		WriteOut(MSG_Get("SHELL_FILE_NOT_FOUND"),
+		         parsed.destination.c_str());
+		return;
+	}
+	std::string canonical_destination = temp;
+
+	// Command uses dta so set it to our internal dta
+	const RealPt save_dta = dos.dta();
+	dos.dta(dos.tables.tempdta);
+	DOS_DTA dta(dos.dta());
+
+	// Each raw source can be a wildcard (*.*) containing multiple real
+	// sources.
+	std::vector<std::string> final_sources;
+	for (const std::string& source : parsed.sources) {
+		if (!DOS_FindFirst(source.c_str(), 0xffff & ~DOS_ATTR_VOLUME)) {
+			WriteOut(MSG_Get("SHELL_FILE_NOT_FOUND"), source.c_str());
+			continue;
+		}
+		if (!DOS_Canonicalize(source.c_str(), temp)) {
+			WriteOut(MSG_Get("SHELL_FILE_NOT_FOUND"), source.c_str());
+			continue;
+		}
+		std::string source_dir = temp;
+		source_dir = source_dir.substr(0, source_dir.rfind(DosSeparator) + 1);
+		do {
+			DOS_DTA::Result result = {};
+			dta.GetResult(result);
+			if (result.name != CurrentDirectory &&
+			    result.name != ParentDirectory) {
+				final_sources.emplace_back(source_dir + result.name);
+			}
+		} while (DOS_FindNext());
+	}
+
+	// Done with DTA. Restore it.
+	dos.dta(save_dta);
+
+	uint16_t destination_attr = 0;
+	bool destination_exists = DOS_GetFileAttr(canonical_destination.c_str(),
+	                                          &destination_attr);
+	bool dest_is_dir       = false;
+
+	if (destination_exists) {
+		if (destination_attr & DOS_ATTR_DIRECTORY) {
+			dest_is_dir = true;
+		} else if (final_sources.size() > 1) {
+			WriteOut(MSG_Get("SHELL_CMD_MOVE_MULTIPLE_TO_SINGLE"));
+			return;
+		}
+	} else if (final_sources.size() > 1) {
+		dest_is_dir = true;
+		if (!DOS_MakeDir(canonical_destination.c_str())) {
+			WriteOut(MSG_Get("SHELL_CMD_MKDIR_ERROR"),
+			         canonical_destination.c_str());
+			return;
+		}
+	}
+
+	if (dest_is_dir && canonical_destination.back() != DosSeparator) {
+		canonical_destination.push_back(DosSeparator);
+	}
+
+	for (const std::string& source : final_sources) {
+		std::string final_destination = canonical_destination;
+		if (dest_is_dir) {
+			final_destination += source.substr(
+			        source.rfind(DosSeparator) + 1);
+		}
+		if (source == final_destination) {
+			WriteOut(MSG_Get("SHELL_FILE_EXISTS"),
+			         final_destination.c_str());
+			continue;
+		}
+		// Overwrite existing file
+		if (destination_exists && !dest_is_dir) {
+			// Condition should be checked for previously
+			assert(final_sources.size() == 1);
+			if (!DOS_UnlinkFile(final_destination.c_str())) {
+				WriteOut(MSG_Get("SHELL_CMD_DEL_ERROR"),
+				         final_destination.c_str());
+				return;
+			}
+		}
+		// If same drive, do a rename. Otherwise it needs a copy + delete.
+		if (source[0] == final_destination[0]) {
```

Comment:
This would describe the intent better:

```cpp
const auto src_drive_letter = source[0];
const auto dest_drive_letter = final_destination[0];

if (src_drive_letter == dest_drive_letter) {
...
```

Reply:
Fixed